src/sx/src/injectRuntimeStyles.js

```diff
@@ -58,7 +59,10 @@ export function injectRuntimeKeyframes(css: string, name: string) {
   };
 
   if (hasStyleRule(matchFunction) === false) {
-    styleSheet.insertRule(css, styleSheet.cssRules.length);
+    const rules = compile(serialize(compile(css), middleware([prefixer, stringify]))) ?? [];
+    rules.forEach((rule) => {
+      styleSheet.insertRule(serialize([rule], stringify), styleSheet.cssRules.length);
```

Comment:
I don't want to discourage you from these efforts but at the same time, I don't want to leave this without mentioning my concern: isn't `serialize(compile(serialize(compile(css))))` a bit too much? 🤔 Have you thought about this @michalsanger?

Reply:
Yes that was ugly, the api of Stylis is a little bit strange. Please check again.